src/averageAges.js

```diff
@@ -55,7 +73,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const child = people.filter(children => onlyWithSon
+    ? people.some(
+      person => children.mother === person.name
+    )
+    && children.sex === 'm'
+    : people.some(
+      person => children.mother === person.name
+    )
+  );
+
+  const averageAgeDiff = child.reduce(
+    (sum, children) => sum + (
+      children.born - people.find(
+        mom => children.mother === mom.name).born),
+    0)
+    / child.length;
```

Comment:
agree it looks way much readable
```suggestion
  const ageDifferencesSum = child.reduce((sum, children) => {
    const mother = people.find(person => children.mother === person.name);
    const ageDifference = children.born - mother.born;
      
    return sum + ageDifference;
  });
  
  const averageAgeDifference = ageDifferencesSum / children.length;
```